File: TOMIMSV4/Entity/Mantenimientos/RoadVendedor/clsBeRoad_p_vendedor.vb
Public Class clsBeRoad_p_vendedor
    Implements ICloneable
    Implements IDisposable

    Public Property IdVendedor() As Integer = 0
    Public Property Codigo() As String = ""
    Public Property Nombre() As String = ""
    Public Property Clave() As String = ""
    Public Property Ruta() As String = ""
    Public Property Nivel() As Integer = 0
    Public Property Nivelprecio() As Integer = 0
    Public Property Bodega() As String = ""
    Public Property Subbodega() As String = ""
    Public Property Cod_vehiculo() As String = ""
    Public Property Liquidando() As String = ""
    Public Property Ultima_fecha_liq() As Date = Date.Now
    Public Property Bloqueado() As Boolean = False
    Public Property Devolucion_sap() As Integer = 0
    Public Property IdRuta As Integer = 0

    Sub New()
    End Sub

    Sub New(ByRef IdVendedor As Integer, ByVal codigo As String, ByVal nombre As String, ByVal clave As String, ByVal ruta As String, ByVal nivel As Integer, ByVal nivelprecio As Integer, ByVal bodega As String, ByVal subbodega As String, ByVal cod_vehiculo As String, ByVal liquidando As String, ByVal ultima_fecha_liq As Date, ByVal bloqueado As Boolean, ByVal devolucion_sap As Integer)
        Me.IdVendedor = IdVendedor
        Me.Codigo = Codigo
        Me.Nombre = Nombre
        Me.Clave = Clave
        Me.Ruta = Ruta
        Me.Nivel = Nivel
        Me.Nivelprecio = Nivelprecio
        Me.Bodega = Bodega
        Me.Subbodega = Subbodega
        Me.Cod_vehiculo = Cod_vehiculo
        Me.Liquidando = Liquidando
        Me.Ultima_fecha_liq = Ultima_fecha_liq
        Me.Bloqueado = Bloqueado
        Me.Devolucion_sap = Devolucion_sap
    End Sub

    Public Function Clone() As Object Implements System.ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function

#Region "IDisposable Support"
    Private disposedValue As Boolean ' To detect redundant calls

    ' IDisposable
    Protected Overridable Sub Dispose(disposing As Boolean)
        If Not disposedValue Then
            If disposing Then
                ' TODO: dispose managed state (managed objects).
            End If

            ' TODO: free unmanaged resources (unmanaged objects) and override Finalize() below.
            ' TODO: set large fields to null.
        End If
        disposedValue = True
    End Sub

    ' TODO: override Finalize() only if Dispose(disposing As Boolean) above has code to free unmanaged resources.
    'Protected Overrides Sub Finalize()
    '    ' Do not change this code.  Put cleanup code in Dispose(disposing As Boolean) above.
    '    Dispose(False)
    '    MyBase.Finalize()
    'End Sub

    ' This code added by Visual Basic to correctly implement the disposable pattern.
    Public Sub Dispose() Implements IDisposable.Dispose
        ' Do not change this code.  Put cleanup code in Dispose(disposing As Boolean) above.
        Dispose(True)
        ' TODO: uncomment the following line if Finalize() is overridden above.
        ' GC.SuppressFinalize(Me)
    End Sub
#End Region

End Class

File: WMS.EntityCore/Road/clsBeRoad_p_vendedor.cs
namespace WMS.EntityCore.Road
{
    using System;

    public class clsBeRoad_p_vendedor : ICloneable, IDisposable
    {
        public int IdVendedor { get; set; } = 0;
        public string Codigo { get; set; } = "";
        public string Nombre { get; set; } = "";
        public string Clave { get; set; } = "";
        public string Ruta { get; set; } = "";
        public int Nivel { get; set; } = 0;
        public int Nivelprecio { get; set; } = 0;
        public string Bodega { get; set; } = "";
        public string Subbodega { get; set; } = "";
        public string Cod_vehiculo { get; set; } = "";
        public string Liquidando { get; set; } = "";
        public DateTime Ultima_fecha_liq { get; set; } = DateTime.Now;
        public bool Bloqueado { get; set; } = false;
        public int Devolucion_sap { get; set; } = 0;
        public int IdRuta { get; set; } = 0;

        public clsBeRoad_p_vendedor() { }

        public clsBeRoad_p_vendedor(int IdVendedor, string codigo, string nombre, string clave, string ruta, int nivel, int nivelprecio, string bodega, string subbodega, string cod_vehiculo, string liquidando, DateTime ultima_fecha_liq, bool bloqueado, int devolucion_sap)
        {
            this.IdVendedor = IdVendedor;
            this.Codigo = codigo;
            this.Nombre = nombre;
            this.Clave = clave;
            this.Ruta = ruta;
            this.Nivel = nivel;
            this.Nivelprecio = nivelprecio;
            this.Bodega = bodega;
            this.Subbodega = subbodega;
            this.Cod_vehiculo = cod_vehiculo;
            this.Liquidando = liquidando;
            this.Ultima_fecha_liq = ultima_fecha_liq;
            this.Bloqueado = bloqueado;
            this.Devolucion_sap = devolucion_sap;
        }

        public object Clone()
        {
            return this.MemberwiseClone();
        }

        private bool disposedValue = false;

        protected virtual void Dispose(bool disposing)
        {
            if (!disposedValue)
            {
                if (disposing)
                {
                }
                disposedValue = true;
            }
        }

        public void Dispose()
        {
            Dispose(true);
            GC.SuppressFinalize(this);
        }
    }
}